orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStates.java

```diff
@@ -204,19 +207,41 @@ private static void validateRunningInfoResultNotNull(RunningJobsResponse running
         }
     }
 
-    public static MinionIpAddressesResponse collectMinionIpAddresses(Retry retry, SaltConnector sc) {
-        return retry.testWith1SecDelayMax5Times(() -> {
-            try {
-                return collectMinionIpAddresses(sc);
-            } catch (RuntimeException e) {
-                LOGGER.error("Collecting minion IP addresses failed", e);
-                throw new Retry.ActionFailedException("Collecting minion IP addresses failed", e);
+    public static Set<String> collectMinionIpAddresses(Retry retry, SaltConnector sc) {
```

Comment:
This method is quite large and has several catches, conditions, etc, so maybe it is better to split into at least 2 methods to increase readability

Reply:
You are right, I will refactor this.